go.mod

```diff
@@ -14,6 +14,7 @@ require (
 	github.com/hashicorp/vault-plugin-secrets-kv v0.9.0
 	github.com/hashicorp/vault/api v1.1.2-0.20210713235431-1fc8af4c041f
 	github.com/hashicorp/vault/sdk v0.2.2-0.20210825150427-9b1f4d486f5d
+	github.com/imdario/mergo v0.3.12
```

Comment:
Can we avoid that new dep to do the merging?

Reply:
The reasoning is simple, let's try not to add a new dep if we don't need to. Maybe we have a dep that does it already. I just looked a `mergo` more closely and it's rather small and it's only pulling `gopkg.in/yaml.v2` so that's ok.

Reply:
I understand the reason now, Thanks :)